### DISCUSSION TITLE
HX711 doesn't report weight via MQTT correctly

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#18592] [Q&A]
I use the HX711 driver for reading weight and to send it to mqtt.

But the sensor value is mostly send when the value is 0 again and it seems this code causes this issue:

https://github.com/arendst/Tasmota/blob/e9d1e8c7250d89a24ade0c42a64731d6c492bbb2/tasmota/tasmota_xsns_sensor/xsns_34_hx711.ino#L433

It seems this code want's 2 readings to be the same, but as the value is probably in grams it won't often be the same in 2 readings (with a scale measuring up to 140kg).

Am I right and need this code to be improved?
I use the latest development version .

**Comment by arendst** [comment]
I think that's where weight_change comes in.

Try to play with command ``Sensor34 9 <x>``. See file regarding syntax.

**Reply by jeroenst** [reply]
I did, but it doesn't work unless the measurement is exactly the same which is not possible when weighting kilograms due to the noise.

**Comment by jeroenst** [comment]
PR that fixes this issue: https://github.com/arendst/Tasmota/pull/18600
